tests/notebook-demo.ipynb

```diff
@@ -0,0 +1,1725 @@
+    "    \"veda-tags.json\",\n",
+    "    \"veda-attr-defaults.json\",\n",
+    "    regions=\"\",\n",
+    "    include_dummy_imports=True,\n",
+    ")\n",
+    "\n",
+    "model = xt.read_xl(\n",
+    "    inputs=[input_dir],\n",
+    "    output_dir=output_dir,\n",
+    "    config=config,\n",
+    "    no_cache=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "d7cad94d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>process</th>\n",
+       "      <th>region</th>\n",
+       "      <th>year</th>\n",
+       "      <th>limtype</th>\n",
+       "      <th>attribute</th>\n",
+       "      <th>value</th>\n",
+       "      <th>source_filename</th>\n",
+       "      <th>module_type</th>\n",
+       "      <th>module_name</th>\n",
+       "      <th>commodity</th>\n",
+       "      <th>timeslice</th>\n",
+       "      <th>other_indexes</th>\n",
+       "      <th>year2</th>\n",
+       "      <th>cg</th>\n",
+       "      <th>currency</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>169</th>\n",
+       "      <td>NaN</td>\n",
+       "      <td>REG1</td>\n",
+       "      <td>2005</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>COM_IE</td>\n",
+       "      <td>0.9</td>\n",
+       "      <td>SysSettings</td>\n",
+       "      <td>syssettings</td>\n",
+       "      <td>SYSSETTINGS</td>\n",
+       "      <td>ELC</td>\n",
+       "      <td>ANNUAL</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>NaN</td>\n",
+       "      <td>MEURO05</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "    process region  year limtype attribute value source_filename  module_type  \\\n",
+       "169     NaN   REG1  2005     NaN    COM_IE   0.9     SysSettings  syssettings   \n",
+       "\n",
+       "     module_name commodity timeslice other_indexes year2   cg currency  \n",
+       "169  SYSSETTINGS       ELC    ANNUAL           NaN   NaN  NaN  MEURO05  "
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model.attributes.query('attribute == \"COM_IE\"')"
```

Comment:
I guess if this workflow looks okay at a high level, the next step would be to modify COM_IE in `model.attributes` and re-run GAMS to check that the solution changes as expected?